network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgServerApi.kt

```diff
@@ -44,43 +45,54 @@ class RealWgServerApi @Inject constructor(
-            RegisterKeyBody(publicKey = publicKey, server = selectedServer)
-        } else if (userPreferredLocation != null) {
-            if (userPreferredLocation.cityName != null) {
-                RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode, city = userPreferredLocation.cityName)
+        val registerKeyBody = try {
+            // This bit of code gets all possible egress servers which should be order by proximity, caches them for internal builds and then
+            // returns the closest one or null if list is empty
+            val selectedServer = wgVpnControllerService.getServers().map { it.server }
+                .also { fetchedServers ->
+                    logcat { "Fetched servers ${fetchedServers.map { it.name }}" }
+                    serverDebugProvider.cacheServers(fetchedServers)
+                }
+                .map { it.name }
+                .firstOrNull { serverName ->
+                    serverDebugProvider.getSelectedServerName()?.let { userSelectedServer ->
+                        serverName == userSelectedServer
+                    } ?: false
+                }
+
+            logcat { "Register key in $selectedServer" }
+            val userPreferredLocation = netNetpEgressServersProvider.updateServerLocationsAndReturnPreferred()
+            if (selectedServer != null) {
+                RegisterKeyBody(publicKey = publicKey, server = selectedServer)
+            } else if (userPreferredLocation != null) {
+                if (userPreferredLocation.cityName != null) {
+                    RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode, city = userPreferredLocation.cityName)
+                } else {
+                    RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode)
+                }
             } else {
-                RegisterKeyBody(publicKey = publicKey, country = userPreferredLocation.countryCode)
+                RegisterKeyBody(publicKey = publicKey, server = "*")
             }
-        } else {
+        } catch (e: Exception) {
             RegisterKeyBody(publicKey = publicKey, server = "*")
         }
 
         return wgVpnControllerService.registerKey(registerKeyBody)
             .run {
-                logcat { "Register key in $selectedServer" }
-                logcat { "Register key returned ${this.map { it.server.name }}" }
-                val server = this.firstOrNull()?.toWgServerData()
-                logcat { "Selected Egress server is $server" }
-                server
+                if (isSuccessful) {
+                    logcat { "Register key returned ${this.body()?.map { it.server.name }}" }
+                    val server = this.body()?.firstOrNull()?.toWgServerData()
+                    logcat { "Selected Egress server is $server" }
+                    netpRepository.disabledDueToAccessRevoked = false
```

Comment:
As discussed, we can try to move the `403` check into the `NetpWaitlistRequestInterceptor` and leave this class as is